uiowa/uiowa
# Create site split for iowasummerwritingfestival.uiowa.edu.

Creation of site split for ISWF.

Resolves #5931
Resolves #5886 

```
ddev blt ds --site=iowasummerwritingfestival.uiowa.edu && ddev drush @iowasummerwritingfestival.local config-split:activate event -y && ddev drush @iowasummerwritingfestival.local en better_exposed_filters -y && ddev drush @iowasummerwritingfestival.local config:import --source ../config/sites/iowasummerwritingfestival.uiowa.edu --partial -y && ddev drush @iowasummerwritingfestival.local cim -y
```

## To Test

### Testing ISWF
- SETUP  
    - `ddev composer install && ddev blt frontend`
    - `ddev blt ds --site=iowasummerwritingfestival.uiowa.edu`
    - `ddev drush @iowasummerwritingfestival.local config-split:activate event -y`
    - `ddev drush @iowasummerwritingfestival.local en better_exposed_filters`
    - `ddev drush @iowasummerwritingfestival.local config:import --source ../config/sites/iowasummerwritingfestival.uiowa.edu --partial -y && ddev drush @iowasummerwritingfestival.local cim -y`
- TESTING
   - Create a new local event and populate with all relevant fields (this includes featured image, short description in sidebar)
      - The genre field should display in two column format. 
   - Ensure that our ISWF custom fields are rendering correctly (event type, genre, and registration link).
   - Ensure that the date display is using the smartdate Default - Long date format.
   - Ensure that the genre and event type fields contain various options and are not empty.
   - Investigate Console display of code, and search for ld+json. Ensure that registration link is mapped to schema.org via the the offers property (will appear in one of the two ld+json sections.
   - Create additional events that have varying empty relevant fields (event type, genre, registration link, short description) and reference original event to test the sub event addition. These new "sub" events should show in "Schedule" region of the first created event's display.
      - The subevent teasers under Schedule should show Presenters comma-separated list.
   - Add or edit a person entity via Content, ensure that 'Presenter' is an option under Person Types.
   - Sticky events, if part of the returned results, should show atop the events_list_block events and sub-events views.

### Testing Default
- SETUP  
    - `ddev composer install && ddev blt frontend`
    - `ddev blt ds`
    - `ddev drush @default.local config-split:activate event -y && ddev drush @default.local cim -y`
- TESTING
    - Follow a similar strategy to the Testing list above for ISWF and ensure this does not break default implementation.

Initial work:

- [x] Moved presenters above description in default event layout.
- [x] Added **field_event_type** single value field. Populated with 'Workshop' to avoid errors preventing future work.
- [x] Added **field_event_genre** field unlimited values field. Populated with previous genres from existing site to avoid errors and test display.
- [x] Added **field_registration_link** field for internal/external links, no title.
- [x] Styled **field_registration_link** field as a button in second column.
- [x] Created 'Default long date' smart date format.
- [x] Changed dates on events view and display layout to 'Default long date' version over 'default'.
- [x] populate field_event_type with allowed values function from custom site module.
- [x] populate field_event_genre with allowed values function from custom site module.
- [x] add field_registration_link field value to offers property in schema.org structure.
- [x] field_registration_link button needs moved above bordered block or fix styling to fit inside current region.
- [x] added 'presenter' person type with performer machine_name to match schema in case we change the label at a later time. (added in single push so can remove if later determined)

ToDo (not complete list):

- [x] Change field_event_genre form display to two column list like Person Type (may not be necessary).
- [x] ~field_event_genre list will not show as comma-delimited in view display, changed back to list for now.~ Removed per client feedback.
- [x] ~Remove field_event_type from view display~ (not necessary)
- [x] Set order of sidebar "meta" information to Location, Virtual Link, Source Link (combination of sitenow_event feature split and this site split)
- [x] Add spacing between meta field displays
- [x] Show event image as widescreen rather than square
- [x]  Update sidebar to use new card template with view that generates a card so that we can separate out styles and code. remove old styling.
- [x] Do not show event image if one is not available, change default not shown away from empty person image
- [x] Update genre list based on customer feedback. Verified with client on 11-14.
- [x] Update event type list based on customer feedback. Verified with client on 11-14.
- [x] Add presenters to sub event view blocks between date and description

### Lazy setup commands
<details>
<summary>with ddev</summary>

```
ddev composer install && ddev blt ds --site=iowasummerwritingfestival.uiowa.edu && ddev drush @iowasummerwritingfestival.local config-split:activate event -y && ddev drush @iowasummerwritingfestival.local en better_exposed_filters -y && ddev drush @iowasummerwritingfestival.local config:import --source ../config/sites/iowasummerwritingfestival.uiowa.edu --partial -y && ddev drush @iowasummerwritingfestival.local cim -y && ddev blt frontend && ddev drush @iowasummerwritingfestival.local cr
```

</details>

<details>
<summary>without ddev</summary>

```
composer install && blt ds --site=iowasummerwritingfestival.uiowa.edu && drush @iowasummerwritingfestival.local config-split:activate event -y && drush @iowasummerwritingfestival.local en better_exposed_filters -y && drush @iowasummerwritingfestival.local config:import --source ../config/sites/iowasummerwritingfestival.uiowa.edu --partial -y && drush @iowasummerwritingfestival.local cim -y && blt frontend && drush @iowasummerwritingfestival.local cr
```

</details>
